agenta-cli/sdk.py

```diff
@@ -0,0 +1,652 @@
+
+# 2.a.ii. existing variant
+variant: Variant = ag.variants.create(
+    variant_reference="...",
+    variant_slug="...",
+)
+
+# 2.b.i. existing variant
+variant: Variant = ag.variants.create(
+    variant="...",
+)
+
+# 2.b.ii. existing variant
+variant: Variant = ag.variants.create(
+    variant="...",
+    variant_slug="...",
+)
+
+##### FETCH #####
+
+# 1. existing variant
+variant: Variant = ag.variants.fetch(
+    variant_reference="...",
+)
+
+# 2. existing environment
+variant: Variant = ag.variants.fetch(
+    environment_reference="...",
+)
+
+##### COMMIT #####
+
+# 1.a. existing variant
+variant: Variant = ag.variants.commit(
+    variant_reference="...",
+)
+
+# 1.b. existing variant
+variant: Variant = ag.variants.commit(
+    variant="...",
+)
+
+# 2.a. existing configuration
+variant: Variant = ag.variants.commit(
+    variant_slug="...",
+    configuration_reference="...",
+)
+
+# 2.b. existing configuration
+variant: Variant = ag.variants.commit(
+    variant_slug="...",
+    configuration="...",
+)
+
+# 3. new variant url
+variant: Variant = ag.variants.commit(
+    variant_slug="...",
+    variant_url="...",
+)
+
+########## ENVIRONMENTS ##########
+
+ag.environments = EnvironmentsClient(...)
+
+##### CREATE #####
+
+# 1.a.i. existing variant
+environment: Environment = ag.prompts.deploy(
+    variant_reference="...",
+)
+
+# 1.a.ii. existing variant
+environment: Environment = ag.prompts.deploy(
+    variant_reference="...",
+    environment_slug="...",
+)
+
+# 1.b.i. existing variant
+environment: Environment = ag.prompts.deploy(
+    variant="...",
+)
+
+# 1.b.ii. existing variant
+environment: Environment = ag.prompts.deploy(
+    variant="...",
+    environment_slug="...",
+)
+
+# 2.a.i. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment_reference="...",
+)
+
+# 2.a.ii. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment_reference="...",
+    environment_slug="...",
+)
+
+# 2.b.i. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment="...",
+)
+
+# 2.b.ii. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment="...",
+    environment_slug="...",
+)
+
+##### FETCH #####
+
+# 1. existing environment
+environment: Environment = ag.environments.fetch(
+    environment_reference="...",
+)
+
+##### DEPLOY #####
+
+# 1.a. existing variant
+environment: Environment = ag.prompts.deploy(
+    variant_reference="...",
+)
+
+# 1.b. existing variant
+environment: Environment = ag.prompts.deploy(
+    variant="...",
+)
+
+# 2.a. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment_slug="...",
+    environment_reference="...",
+)
+
+# 2.b. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment_slug="...",
+    environment="...",
+)
+
+
+########## PROMPTS ##########
+
+##### CREATE #####
+
+# 1.a.i. new prompt
+prompt: Prompt = ag.prompts.create(
```

Comment:
does this create a configuration (just the parameters), or also the app/base_id and variant

Reply:
This one _would_ create the app and base.
_**But**_, we said that we wouldn't implement it now due to the 30-sec delay in creating an service. Correct ?

Reply:
yes